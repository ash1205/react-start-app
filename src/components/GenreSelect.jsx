import React from 'react';
import { useState } from 'react';
import { allGenres } from '../lib/constants';
import ButtonGroup from 'react-bootstrap/ButtonGroup';
import ToggleButton from 'react-bootstrap/ToggleButton';



function GenreSelect({ genres, selectedGenre, onSelect }) {
  const [genreValue, setGenreValue] = useState('1');

    return (
      <div>
      <ButtonGroup>
        {allGenres.map((genre, idx) => (
          <ToggleButton
            key={idx}
            id={`genre-${idx}`}
            type="radio"
            variant={genre.variant}
            name="genre"
            value={genre.value}
            checked={genreValue === genre.value}
            onChange={(event) => setGenreValue(event.currentTarget.value)}
          >
            {genre.name}
          </ToggleButton>
        ))}
      </ButtonGroup>
      </div>
    );
  }

export default GenreSelect;
