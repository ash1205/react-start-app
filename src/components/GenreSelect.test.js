import React from "react";
import { render, fireEvent } from "@testing-library/react";
import GenreSelect from "./GenreSelect";

const mockHandleGenreChanges = jest.fn();

const mockGenres = [
  { name: "Action", value: "action", variant: "primary" },
  { name: "Comedy", value: "comedy", variant: "success" },
];

describe("GenreSelect Component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <GenreSelect allGenres={mockGenres} handleGenreChanges={mockHandleGenreChanges} />
    );
    expect(container).toMatchSnapshot();
  });

  it("calls handleGenreChanges when a genre is selected", () => {
    const { getByLabelText } = render(
      <GenreSelect allGenres={mockGenres} handleGenreChanges={mockHandleGenreChanges} />
    );

    fireEvent.click(getByLabelText("Action"));

    expect(mockHandleGenreChanges).toHaveBeenCalledWith("Action");
  });

});
