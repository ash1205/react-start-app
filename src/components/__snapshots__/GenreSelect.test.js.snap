// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreSelect Component renders correctly 1`] = `
<div>
  <div>
    <div
      class="btn-group"
      role="group"
    >
      <input
        autocomplete="off"
        class="btn-check"
        id="genre-0"
        name="genre"
        type="radio"
        value="Action"
      />
      <label
        class="btn btn-primary"
        for="genre-0"
        tabindex="0"
      >
        Action
      </label>
      <input
        autocomplete="off"
        class="btn-check"
        id="genre-1"
        name="genre"
        type="radio"
        value="Comedy"
      />
      <label
        class="btn btn-success"
        for="genre-1"
        tabindex="0"
      >
        Comedy
      </label>
    </div>
  </div>
</div>
`;
